lib/streamlit/commands/execution_control.py

```diff
@@ -87,3 +90,49 @@ def experimental_rerun() -> NoReturn:
     # be seen.
     _LOGGER.warning(msg)
     rerun()
+
+
+@gather_metrics("switch_page")
+def switch_page(page: str) -> NoReturn:
+    """Switch the current programmatically page in a multi-page app.
+    When `st.switch_page()` is called with a page, the current page script is halted
+    and the requested page script will be queued to run from the top.
+    Parameters
+    ----------
+    page: str
+        The file path, relative to the main script, of the page to switch to.
+    """
+
+    ctx = get_script_run_ctx()
+
+    query_string = ""
+    ctx_main_script = ""
+    if ctx:
+        query_string = ctx.query_string
+        ctx_main_script = ctx.main_script_path
+
+    main_script_path = os.path.join(os.getcwd(), ctx_main_script)
+    main_script_directory = os.path.dirname(main_script_path)
+
+    if page.startswith("./"):
+        page = page[2:]
+    elif page.startswith("/"):
+        page = page[1:]
```

Comment:
It seems to me this is a convenience that avoids two things:

- It avoids the issue of `path.join` interpreting a path from the root directory (`path.join("/to/file", "/new/path") => /new/path`)
- It helps with user confusion to include the `/` (if we didn't have it, it would error below).

If my understanding is correct, then we should put a comment explaining the purpose of this check.

also, TIL, you can do `page = page.strip("/")` and that will remove the preceding and ending `/`